File: app/src/main/java/com/mundi/databaseadmin/database/QueryData.java
package com.mundi.databaseadmin.database;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONObject;

public class QueryData {
    private static final String TAG = "QueryData";

    private BackendConnection backendConnection;
    public QueryData(String uri, String databasename, String tablename) {
        Log.d(TAG, "QueryData()");
        String cmd = "querydata&dbname=" + databasename + "&tablename=" + tablename;
        backendConnection = new BackendConnection(uri, cmd);
    }

    public JSONArray getData() {
        Log.d(TAG, "getData()");
        return backendConnection.getJSONArray();
    }

}

File: app-kotlin/src/main/java/com/mundi/databaseadmin/database/QueryData.kt
package com.mundi.databaseadmin.database

import android.util.Log
import org.json.JSONArray

class QueryData(uri: String?, databasename: String, tablename: String) {
    private val backendConnection: BackendConnection
    val data: JSONArray
        get() {
            Log.d(TAG, "getData()")
            return backendConnection.jsonArray
        }

    companion object {
        private const val TAG = "QueryData"
    }

    init {
        Log.d(TAG, "QueryData()")
        val cmd = "querydata&dbname=$databasename&tablename=$tablename"
        backendConnection = BackendConnection(uri, cmd)
    }
}